lib/lf/fe/is_mesh_function.h

```diff
@@ -0,0 +1,65 @@
+/**
+ * @file
+ * @brief Defines a compile time constant that can be used to check if a given
+ *        type fulfills the MeshFunction concept.
+ * @author Raffael Casagrande
+ * @date   2018-12-15 03:49:01
+ * @copyright MIT License
+ */
+
+#ifndef __7e4ffaa81e244723acbfbbaea68e03b1
+#define __7e4ffaa81e244723acbfbbaea68e03b1
+
+#include <lf/mesh/mesh.h>
+#include <type_traits>
+
+namespace lf::fe {
+namespace internal {
```

Comment:
Can you please add a few inline comments on the purpose of this code. I do not completely understand the template magic here. 

Have you got a general reference where such template coding patterns are explained.

Reply:
I'm not sure if there is a standard pattern for coding this sort of stuff. This is out of the top of my head. It's not nice, but that's the reason it resides in the `internal` namespace. I don't think we need to put comments there because normal users are not supposed to read that. There are many things happening there and describing them all in comments is not so easy.

Reply:
OK, then hide it.